tests/test_data_management.py

```diff
@@ -155,3 +155,55 @@ def test_load_results(self, mock_file: MagicMock, mock_makedirs: MagicMock):
+
+    original_platform = ql.build_platform(Galadriel.runcard)
+    path = save_platform(path="./test.yml", platform=original_platform)
+    saved_platform = ql.build_platform(path)
+    os.remove(path)
+
+    def test_long_decimal_in_serialization(self):
+        """Test that long decimals are correctly dumped in serialization."""
+        original_phase = self.original_platform.gates_settings.gates["Y(0)"][0].pulse.phase
+        original_phase_from_dict = self.original_platform.to_dict()["gates_settings"]["gates"]["Y(0)"][0]["pulse"][
+            "phase"
+        ]
+        saved_phase = self.saved_platform.gates_settings.gates["Y(0)"][0].pulse.phase
+        saved_phase_from_dict = self.saved_platform.to_dict()["gates_settings"]["gates"]["Y(0)"][0]["pulse"]["phase"]
+
+        assert original_phase == original_phase_from_dict == saved_phase == saved_phase_from_dict == 1.5707963267948966
+
+    def test_underscore_notation_in_serialization(self):
+        """Test that underscore notations are correctly dumped in serialization."""
+        original_if = self.original_platform.instruments.elements[0].awg_sequencers[0].intermediate_frequency
+        original_phase_from_dict = self.original_platform.to_dict()["instruments"][0]["awg_sequencers"][0][
+            "intermediate_frequency"
+        ]
+        saved_if = self.saved_platform.instruments.elements[0].awg_sequencers[0].intermediate_frequency
+        saved_phase_from_dict = self.saved_platform.to_dict()["instruments"][0]["awg_sequencers"][0][
+            "intermediate_frequency"
+        ]
+
+        assert original_if == original_phase_from_dict == saved_if == saved_phase_from_dict == 100_000_000
+
+    def test_scientific_notation_in_serialization(self):
+        """Test that scientific notations are correctly dumped in serialization."""
+        original_lo_freq = self.original_platform.instruments.elements[2].frequency
+        original_lo_freq_from_dict = self.original_platform.to_dict()["instruments"][2]["frequency"]
+        saved_lo_freq = self.saved_platform.instruments.elements[2].frequency
+        saved_lo_freq_from_dict = self.saved_platform.to_dict()["instruments"][2]["frequency"]
+
+        assert original_lo_freq == original_lo_freq_from_dict == saved_lo_freq_from_dict == saved_lo_freq == 7.24730e09
+
+    def test_None_in_serialization(self):
+        """Test that None's are correctly dumped in serialization."""
+        original_firmware = self.original_platform.instruments.elements[4].firmware
+        original_firmware_from_dict = self.original_platform.to_dict()["instruments"][4]["firmware"]
+        saved_firmware = self.saved_platform.instruments.elements[4].firmware
+        saved_firmware_from_dict = self.saved_platform.to_dict()["instruments"][4]["firmware"]
+
+        assert original_firmware == original_firmware_from_dict == saved_firmware == saved_firmware_from_dict
+        assert saved_firmware is None
```

Comment:
By the way, could we merge all of these tests in a single one?

Just doing the following assert would suffice, right?
```python
assert original_platform == saved_platform
```

Reply:
Reduced them by joining them with the already existing tests!